docs/index.md

```diff
@@ -50,155 +55,91 @@ Compose has commands for managing the whole lifecycle of your application:
+Compose preserves all volumes used by your services. When `docker-compose up`
+runs, if it finds any containers from previous runs, it copies the volumes from
+the old container to the new container. This process ensures that any data
+you've created in volumes isn't lost.
 
-    if __name__ == "__main__":
-        app.run(host="0.0.0.0", debug=True)
 
-Next, define the Python dependencies in a file called `requirements.txt`:
+#### Only recreate containers that have changed
 
-    flask
-    redis
+Compose caches the configuration used to create a container. When you
+restart a service that has not changed, Compose re-uses the existing
+containers. Re-using containers means that you can make changes to your
+environment very quickly.
 
-### Create a Docker image
 
-Now, create a Docker image containing all of your app's dependencies. You
-specify how to build the image using a file called
-[`Dockerfile`](http://docs.docker.com/reference/builder/):
+#### Variables and moving a composition to different environments
 
-    FROM python:2.7
-    ADD . /code
-    WORKDIR /code
-    RUN pip install -r requirements.txt
-    CMD python app.py
+> New in `docker-compose` 1.5
 
-This tells Docker to:
+Compose supports variables in the Compose file. You can use these variables
+to customize your composition for different environments, or different users.
+See [Variable substitution](compose-file.md#variable-substitution) for more
+details.
 
-* Build an image starting with the Python 2.7 image.
-* Add the current directory `.` into the path `/code` in the image.
-* Set the working directory to `/code`.
-* Install the Python dependencies.
-* Set the default command for the container to `python app.py`
+Compose files can also be extended from other files using the `extends`
+field in a compose file, or by using multiple files. See [extends](extends.md)
+for more details.
 
-For more information on how to write Dockerfiles, see the [Docker user guide](https://docs.docker.com/userguide/dockerimages/#building-an-image-from-a-dockerfile) and the [Dockerfile reference](http://docs.docker.com/reference/builder/).
 
-You can build the image by running `docker build -t web .`.
+## Common Use Cases
 
-### Define services
+Compose can be used in many different ways. Some common use cases are outlined
+below.
 
-Next, define a set of services using `docker-compose.yml`:
+### Development environments
 
-    web:
-      build: .
-      ports:
-       - "5000:5000"
-      volumes:
-       - .:/code
-    redis:
-      image: redis
+When you're developing software it is often helpful to be able to run the
+application and interact with it. If the application has any service dependencies
+(databases, queues, caches, web services, etc) you need a way to document the
+dependencies, configuration and operation of each. Compose provides a convenient
+format for definition these dependencies (the [Compose file](yml.md)) and a CLI
```

Comment:
Scattering links in text body isn't really as effective as placing key links in clear xrefs at the end of a section or page.  Effectively by putting a link in the middle of your sentence or section you are encouraging a use to change channels in the middle of a thought.  An Xref provides a link and some idea of where the user is going and why.

Also, big honking paragraph...your lead isn't properly tied to your point.  And, um, the last sentence is a bit markety and bit of a stretch.  I think it can replace a lot but not an entire developer guide.

--->

 When you're developing software, the ability to run an application and interact
 with it is crucial. Compose includes a command line tool for starting an isolated environment. Once an environment is running, you can use XXX to interact with it.

 If an application has any service dependencies (databases, queues, caches, web
 services, etc) documenting its the dependencies, configuration, and operation
 is crucial. The Compose file has a readable format which self-documents an
 application's dependencies.  The file can replace a multi-page "developer
 getting started guide" with a single machine readable configuration file and a
 single command `docker-compose up`.

For more detailed information about configuring an application with a Compose file, see [the file's component reference](yml.md).